### ISSUE TITLE
Enable `HelpSearch` card on Dotcom "My Home" for all non-Desktop App layouts

### ISSUE THREAD

**Issue by getdave** [Automattic/wp-calypso#42744]
Following the merge of the client-side code for the new `HelpSearch` support search card for the "My Home" section of Calypso we now need to merge the changes to the layout API to make the new card show up to users.

This Issue tracks what is left to complete before we can consider showing the MVP version fo the card to users.

- [ ] ~merge `code-D43941` having first tested across all layouts and views on testing on Simple and Atomic.~ Closed in favour of approach in https://github.com/Automattic/wp-calypso/pull/42752
- [x] Review https://github.com/Automattic/wp-calypso/pull/42752 to see if it's preferable to approach in  `code-D43941`.
- [x] Doubling check that the card does not show up for Desktop App (ping team member for sign off on diff). See also discussion at `paYJgx-GU-p2`.
- [x] ~[Approval from Design on the Card UI itself](https://github.com/Automattic/wp-calypso/issues/42743).~ @davemart-in was happy to launch and [tweak Design in followup](https://github.com/Automattic/wp-calypso/pull/42980).
